homeassistant/components/meteoclimatic/sensor.py

```diff
@@ -0,0 +1,127 @@
+"""Support for Meteoclimatic sensor."""
+import logging
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import ATTR_ATTRIBUTION
+from homeassistant.core import callback
+from homeassistant.helpers.entity import Entity
+from homeassistant.helpers.typing import HomeAssistantType
+
+from .const import (
+    ATTRIBUTION,
+    DOMAIN,
+    METEOCLIMATIC_COORDINATOR,
+    METEOCLIMATIC_STATION_CODE,
+    METEOCLIMATIC_STATION_NAME,
+    METEOCLIMATIC_UPDATER,
+    SENSOR_TYPE_CLASS,
+    SENSOR_TYPE_ICON,
+    SENSOR_TYPE_NAME,
+    SENSOR_TYPE_UNIT,
+    SENSOR_TYPES,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_LAST_UPDATE = "last_update"
+ATTR_SENSOR_ID = "sensor_id"
+ATTR_STATION_CODE = "station_code"
+ATTR_STATION_NAME = "station_name"
+
+
+async def async_setup_entry(
+    hass: HomeAssistantType, entry: ConfigEntry, async_add_entities
+) -> None:
+    """Set up the Meteoclimatic sensor platform."""
+    hass_data = hass.data[DOMAIN][entry.entry_id]
+
+    async_add_entities(
+        [MeteoclimaticSensor(sensor_type, hass_data) for sensor_type in SENSOR_TYPES],
+        False,
+    )
+
+
+class MeteoclimaticSensor(Entity):
```

Comment:
Extend from CoordinatorEntity, as it implements some basics for the data update coordinator.